comet-java-client/src/main/java/ml/comet/experiment/http/ConnectionUtils.java

```diff
@@ -0,0 +1,124 @@
+    static final String JSON_MIME_TYPE = "application/json";
+
+    /**
+     * Creates GET request to the given endpoint with specified query parameters.
+     *
+     * @param url    the endpoint URL
+     * @param params the request parameters.
+     * @return the GET request.
+     */
+    static Request createGetRequest(@NonNull String url, Map<String, String> params) {
+        RequestBuilder builder = new RequestBuilder();
+        builder.setUrl(url);
+        if (params != null) {
+            params.forEach(builder::addQueryParam);
+        }
+
+        return builder.build();
+    }
+
+    /**
+     * Creates POST request from given file to the specified endpoint.
+     *
+     * @param file   the file to be included in the body parts.
+     * @param url    the URL of the endpoint.
+     * @param params the query parameters of the request.
+     * @return the POST request with specified file.
+     */
+    static Request createPostFileRequest(@NonNull File file, @NonNull String url, Map<String, String> params) {
+        RequestBuilder builder = new RequestBuilder(HttpConstants.Methods.POST);
+        builder
+                .setUrl(url)
+                .setHeader("Content-Type", FORM_MIME_TYPE)
+                .addBodyPart(new FilePart(FILE, file, FORM_MIME_TYPE));
+        if (params != null) {
+            params.forEach(builder::addQueryParam);
+        }
+
+        return builder.build();
+    }
+
+    /**
+     * Creates POST request from given byte array to the specified endpoint.
+     *
+     * @param bytes  the bytes array to include into request.
+     * @param url    the URL of the endpoint.
+     * @param params the query parameters of the request.
+     * @return the POST request with specified byte array as body part.
+     */
+    static Request createPostByteArrayRequest(byte[] bytes, @NonNull String url, Map<String, String> params) {
+        RequestBuilder builder = new RequestBuilder(HttpConstants.Methods.POST);
+        builder
+                .setUrl(url)
+                .setHeader("Content-Type", FORM_MIME_TYPE)
+                .addBodyPart(new ByteArrayPart(FILE, bytes, FORM_MIME_TYPE));
+        if (params != null) {
+            params.forEach(builder::addQueryParam);
+        }
+
+        return builder.build();
+    }
+
+    /**
+     * Creates Request with specified body for given url.
+     */
+    static Request createPostJsonRequest(@NonNull String body, @NonNull String url) {
+        return new RequestBuilder()
+                .setUrl(url)
+                .setHeader("Content-Type", JSON_MIME_TYPE)
+                .setBody(new ByteArrayBodyGenerator(body.getBytes()))
+                .setMethod(HttpConstants.Methods.POST)
+                .build();
+    }
+
+    /**
+     * Returns true if the status code is in [200..300), which means the request was successfully received,
+     * understood, and accepted.
+     */
+    static boolean isResponseSuccessful(int statusCode) {
```

Comment:
its possible to use https://docs.oracle.com/javaee/6/api/index.html to use built in function without impl of http logic ourself

Reply:
My assumption that we are limited by Java 2 SE. The Java API you are mentioned is belonging to the Java 2 Enterprise Edition. 

Should I assume that we are going to use J2EE for the Comet Java SDK client library? I was intentionally trying to avoid any dependencies on J2EE.

Reply:
As a result of our sync up meeting, we decided to leave for now.